selfdrive/boardd/panda.cc

```diff
@@ -124,7 +124,7 @@ std::vector<std::string> Panda::list() {
     if (desc.idVendor == 0xbbaa && desc.idProduct == 0xddcc) {
       libusb_device_handle *handle = NULL;
       libusb_open(device, &handle);
-      unsigned char serial[25];
+      unsigned char serial[26];
```

Comment:
You're not initializing the array to zero, so still no null byte at the end.

Reply:
But since you're using the length when converting to std::string it shouldn't be necessary to have a null byte.